java-squid/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java

```diff
@@ -35,8 +35,16 @@
 
   CompilationUnitTree getTree();
 
+  /**
+   * @deprecated use reportIssue instead to benefit from precise issue location.
+   */
+  @Deprecated
   void addIssue(Tree tree, JavaCheck check, String message);
 
+  /**
+   * @deprecated use reportIssue instead to benefit from precise issue location.
+   */
+  @Deprecated
   void addIssue(Tree tree, JavaCheck check, String message, @Nullable Double cost);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)